kernel/thread/thread.c

```diff
@@ -1055,6 +1100,19 @@ int thd_init(void) {
     strcpy(reaper->label, "[reaper]");
     thd_set_prio(reaper, 1);
 
+    /* Check whether we should set initial CPU time values. */
+    if(perf_cntr_timer_enabled()) {
+        /* Get initial total process CPU time */
+        const uint64_t ns = perf_cntr_timer_ns();
+        /* Set it as our scheduled timestamp */
+        kern->cpu_time.scheduled = ns;
+        /* Also set it as our total CPU time, so that the main
+           thread is credited with consuming this CPU time, otherwise
+           it will be lost and thread totals will not equal process
+           total. */
+        kern->cpu_time.total = ns;
+    }
```

Comment:
Should these be 0'd in the negative case? See the earlier comment about it being really runtime variable or not.

Reply:
Responded.... Good question... should they? What do you think?

Reply:
Dug deeper. On thread creation we do a memset 0, which is why this was working fine for new threads. So it's already 0 initted as a default.